miden-crypto/src/merkle/smt/mod.rs

```diff
@@ -39,6 +39,12 @@ pub const SMT_MAX_DEPTH: u8 = 64;
 // SPARSE MERKLE TREE
 // ================================================================================================
 
+/// Derive a leaf index from the key of the key-value-pair stored
+pub fn word_to_leaf_index(key: &Word) -> LeafIndex<SMT_DEPTH> {
+    let most_significant_felt = key[3];
+    LeafIndex::new_max_depth(most_significant_felt.as_int())
+}
```

Comment:
Is this not already available via `From<Word> for LeafIndex<SMT_DEPTH>`? I think we should be able to do the conversion as:
```rust
let leaf_index = LeafIndex::<SMT_DEPTH>::from(key);
```

Also, I wonder if this means that https://github.com/0xMiden/crypto/issues/515 is already addressed. cc @PhilippGackstatter

Reply:
It is, I missed that one. Thank you!